### PR TITLE
[Improvement] Remove taskQueue and looper in worker

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

Since we have put the `delay` logic from worker to master, there is no need to add task queue in worker, this will cause it's not convenient to judge the task size, and we need to use a thread to loop the task queue.

This pr will remove the extra task queue in worker, directly use the thread pool's queue as task queue.

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/execute/MasterAsyncTaskExecutorThreadPool.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    public boolean removeMasterTaskExecutor(AsyncMasterTaskExecutor asyncMasterTaskExecutor) {
+        return threadPoolExecutor.remove(asyncMasterTaskExecutor);
+    }
+
+    // todo: remove this method
+    ThreadPoolExecutor getThreadPool() {
+        return threadPoolExecutor;
+    }
+}
```
Comment (caishunfeng) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/execute/MasterAsyncTaskExecutorThreadPool.java]: Can you add more comments for it? You added this method but the todo is the remove method, which seems a bit strange.

File: dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java [hunk 2]
```diff
@@ -128,26 +107,4 @@ public void stop(String cause) {
         close(cause);
     }
 
-    public void killAllRunningTasks() {
```
Comment (caishunfeng) [inline dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java:131]: I think we keep this method to make sure all task are canceled when worker down, WDYT?

**Comment by caishunfeng** [comment]
Please add some pr description.

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/execute/MasterAsyncTaskExecutorThreadPool.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/execute/MasterAsyncTaskExecutorThreadPool.java]: Done.

File: dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java (hunk 2 above)
Comment (ruanwenjun) [reply dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java:131]: Ok, I keep this method, and add todo.

**Comment by ruanwenjun** [comment]
> Please add some pr description.

Done

**Review by caishunfeng — APPROVED** [review]
LGTM

**Pull request merged**